conans/model/build_info.py

```diff
@@ -1,224 +1,596 @@
+        self.resdirs = [DEFAULT_RES]  # Directories to find resources, data, etc
+        self.bindirs = [DEFAULT_BIN]  # Directories to find executables and shared libs
+        self.builddirs = [DEFAULT_BUILD]
         self.libs = []  # The libs to link against
+        self.exes = []  # The exes
         self.defines = []  # preprocessor definitions
         self.cflags = []  # pure C flags
         self.cxxflags = []  # C++ compilation flags
         self.sharedlinkflags = []  # linker flags
         self.exelinkflags = []  # linker flags
-        self.rootpath = ""
+        self._rootpath = root_folder
         self.sysroot = ""
+        self.version = None
+        self.description = None
+        # When package is editable, filter_empty=False, so empty dirs are maintained
+        self.filter_empty = True   # FIXME: Should not be part of the public interface
+        self._components = OrderedDict()
+        self.public_deps = []
+        self.configs = {}    # FIXME: Should not be part of the public interface
+        self._default_values = {
+            "includedirs": [DEFAULT_INCLUDE],
+            "libdirs": [DEFAULT_LIB],
+            "bindirs": [DEFAULT_BIN],
+            "resdirs": [DEFAULT_RES],
+            "builddirs": [DEFAULT_BUILD],
+            "srcdirs": [],
+            "libs": [],
+            "exes": [],
+            "defines": [],
+            "cflags": [],
+            "cxxflags": [],
+            "sharedlinkflags": [],
+            "exelinkflags": []
+        }
+
+    @property
+    def rootpath(self):
+        return self._rootpath
+
+    # Compatibility for 'cppflags'
+    @deprecation.deprecated(deprecated_in="1.13", removed_in="2.0",
+                            details="Use 'cxxflags' instead")
+    def get_cppflags(self):
+        return self.cxxflags
+
+    # Compatibility for 'cppflags'
+    @deprecation.deprecated(deprecated_in="1.13", removed_in="2.0",
+                            details="Use 'cxxflags' instead")
+    def set_cppflags(self, value):
+        self.cxxflags = value
+
+    # Old style property to allow deprecation decorators
+    cppflags = property(get_cppflags, set_cppflags)
+
+    def _check_and_clear_default_values(self):
+        for dir_name in self._default_values:
+            dirs_value = getattr(self, dir_name)
+            if dirs_value is not None and dirs_value != self._default_values[dir_name]:
+                msg_template = "Using Components and global '{}' values ('{}') is not supported"
+                raise ConanException(msg_template.format(dir_name, dirs_value))
+            else:
+                self.__dict__[dir_name] = None
+
+    def __getitem__(self, key):
+        self._check_and_clear_default_values()
+        if key not in self._components:
+            self._components[key] = Component(key, self.rootpath)
+        return self._components[key]
+
+    @property
+    def components(self):
+        return self._components
```

Comment:
Should this return ``self._components.items()`` to follow the same scheme as for ``DepsCppInfo::dependencies``??

Reply:
Done!